profiling/src/allocation/mod.rs

```diff
@@ -65,22 +65,36 @@ impl AllocationProfilingStats {
 
         self.next_sampling_interval();
 
-        if let Some(profiler) = Profiler::get() {
-            // Safety: execute_data was provided by the engine, and the profiler doesn't mutate it.
-            unsafe {
-                profiler.collect_allocations(
-                    zend::ddog_php_prof_get_current_execute_data(),
-                    1_i64,
-                    len as i64,
-                )
-            };
-        }
+        // Queue the allocation for safe-point collection
+        REQUEST_LOCALS.with_borrow_mut(|locals| {
+            locals
+                .allocation_interrupt_count
+                .fetch_add(1, Ordering::SeqCst);
+        });
+
+        PENDING_ALLOCATION_SIZE.with_borrow_mut(|pending_size| {
+            // Trigger interrupt only when transitioning from zero to non-zero
+            if *pending_size == 0 {
+                *pending_size += len as u64;
+                if let Some(profiler) = Profiler::get() {
+                    // TODO: If this thread needs to take an allocation sample, calling
+                    // `profiler.trigger_interrupt()` will not only trigger this threads interrupt,
+                    // but all other PHP ZTS threads interrupts as well. The interrupt handler is
+                    // pretty slim, and does not collect a stack trace if there is nothing pending,
+                    // yet we should only trigger an interrupt in the "current" thread.
+                    profiler.trigger_interrupt();
```

Comment:
Even though we could fix it here, I'd say we are also save to ignore this for the time being, as only very very very few people are using ZTS PHP.